pkg/agent/controller/agent.go

```diff
@@ -486,18 +495,10 @@ func (a *Controller) newServiceImport(svcExport *mcsv1a1.ServiceExport) *mcsv1a1
 
 func (a *Controller) getIPsAndPortsForService(service *corev1.Service, siType mcsv1a1.ServiceImportType) (
 	[]string, []mcsv1a1.ServicePort, error) {
-	var mcsPorts = make([]mcsv1a1.ServicePort, 0, len(service.Spec.Ports))
-
-	for _, port := range service.Spec.Ports {
-		mcsPorts = append(mcsPorts, mcsv1a1.ServicePort{
-			Name:     port.Name,
-			Protocol: port.Protocol,
-			Port:     port.Port,
-		})
-	}
+	mcsPorts := a.getPortsForService(service)
 
 	if siType == mcsv1a1.ClusterSetIP {
-		mcsIP := getGlobalIPFromService(service)
+		mcsIP, _, _ := a.getGlobalIP(service)
```

Comment:
Since we already retrieved and parsed the global IP previously, we can pass it into this function to avoid the overhead of repeating it. Also it's possible `getGlobalIP` could fail here where it didn't previously.